docs/modules/clients/pages/java-thin-client.adoc

```diff
@@ -116,23 +116,19 @@ The main areas are around client connections and retry-able operations. Some app
 
 **Handling Client Connection Failure:**
 
-While the client initially tries to connect to one of the members in the
-`ClientNetworkConfig.addressList`, it is possible that not all members are available.
-Instead of giving up, throwing an exception and stopping,
-the client continues to attempt to connect as configured.
-For information on the available configuration, see <<configuring-client-connection-retry, Configuring Client Connection Retry>>.
+While the client is trying to connect initially to one of the members in the
+`ClientNetworkConfig.addressList`, all the members might be not available.
+Instead of giving up, throwing an exception and stopping the client,
+the client retries to connect as configured which is described in the
+<<configuring-client-connection-retry, Configuring Client Connection Retry section>>.
 
-The client executes each operation through the already established connection to the cluster.
-If this connection disconnects or drops, the client tries to reconnect as configured.
+The client executes each operation through the already established connection(s) to the cluster.
+If these connection(s) disconnect or drop, the client tries to reconnect as configured.
 
-The initial connection is established using one of the addresses provided in the <<configuring-address-list, address list>>.
-The client gets the addresses of other members in the cluster from the first connected member.
-
-If you use a <<client-network,discovery mechanism>> to find the initial member for the connection instead of an address list,
-you can use the same property to configure whether the initial member connection uses the private or public address.
-
-For an example of this scenario, refer to
-link:https://docs.hazelcast.com/tutorials/hazelcast-platform-operator-expose-externally[Connect to Hazelcast from Outside Kubernetes, window=_blank] in the Operator documentation.
+If using the `MULTI_MEMBER` cluster routing mode, and the cluster has multiple partition groups defined 
+and the client connection to a partition group fails, connectivity is maintained by failing over to an alternative partition group.
+If the connection is lost, which occurs only if all members of the partition group become unavailable, there is no attempt to retry the connection before failing over to another partition group.
+For further information on client cluster routing modes, see <<client-cluster-routing-modes,Client Cluster Routing Modes>>.
```

Comment:
Does the thin client support cluster routing modes? like MULTI_MEMBER etc?

Reply:
It does but in the newest version. In the verion 5.5 (the one which this doc relates to) it didn't. I have reverted the change and removed the mention of that property. Hope this is OK for now since once this is merged we will submit another PR for the newest version -- in which mentioning MULTI_MEMBER  and ither cluster routing modes will be valid